src/variety/cbblock.js

```diff
@@ -24,10 +24,46 @@ MouseEvent:{
 		}
 	}
 },
+"MouseEvent@dbchoco":{
+	inputModes : {edit:['ice','number','clear'],play:['border','subline']},
+	mouseinput_auto : function(){
+		if(this.puzzle.playmode){
+			if(this.mousestart || this.mousemove){
+				if(this.btn==='left' && this.isBorderMode()){ this.inputborder();}
+				else{ this.inputQsubLine();}
+			}
+		}
+		else if(this.puzzle.editmode){
+			var cell = this.getcell();
+			if(cell.isnull){ return;}
+
+			if(this.mousestart && (this.btn!=='right' || cell===this.cursor.getc())) {
+				this.inputData = -1;
+			}
+
+			if( (this.mousestart && cell!==this.cursor.getc() && this.btn==='right') ||
+				(this.mousemove && this.inputData>=0)) {
+				this.inputIcebarn();
+			}
```

Comment:
at first this felt buggy, but I agree now that it's a neat way to allow the pointer to enter both clues and shading